File: Go/Math/get-smallest-number-in-slice.go
package main

import "fmt"

func main() {
	// Create a slice of random data.
	randomNumbers := []int{0, 1, 2, 3, 4}
	// Get the smallest number in a slice.
	fmt.Println(getSmallestNumberInSlice(randomNumbers))
}

// Get the smallest number in a given slice.
func getSmallestNumberInSlice(slice []int) int {
	var smallestNumber int
	for _, value := range slice {
		if value < smallestNumber {
			smallestNumber = value
		}
	}
	return smallestNumber
}

File: Dart/Math/get_smallest_number_in_slice.dart
import "dart:core";
import "dart:math";

void main() {
  // Create of list of numbers.
  final List<int> randomSlice = <int>[0, 1, 2, 3];
  // Find the smallest number in the given slice.
  print(getSmallestNumberInSlice(randomSlice));
}

// Get the smallest number in a given slice.
int getSmallestNumberInSlice(final List<int> givenSlice) {
  return givenSlice.reduce(min);
}
